File: mobile/app/src/main/java/com/hoangcn/quizbattle/battles/models/SubmitMatchAnswerRequest.java
package com.hoangcn.quizbattle.battles.models;

import java.util.ArrayList;
import java.util.List;

public class SubmitMatchAnswerRequest {
    private final String trackingId;
    private final int questionId;
    private final List<String> answers;

    public SubmitMatchAnswerRequest(String trackingId, int questionId, List<String> answers) {
        this.trackingId = trackingId;
        this.questionId = questionId;
        this.answers = answers == null ? new ArrayList<>() : answers;
    }
}

File: server/Models/Matchs/Requests/SubmitMatchAnswerRequest.cs
namespace Models.Matchs.Requests
{
    public class SubmitMatchAnswerRequest
    {
        public string TrackingId { get; set; } = string.Empty;
        public int QuestionId { get; set; }
        public List<string> Answers { get; set; } = new();
    }
}
